client/landing/stepper/declarative-flow/internals/steps-repository/link-in-bio-post-setup/index.tsx

```diff
@@ -0,0 +1,97 @@
+/* eslint-disable wpcalypso/jsx-classname-namespace */
+import { StepContainer, base64ImageToBlob } from '@automattic/onboarding';
+import { useDispatch } from '@wordpress/data';
+import { createInterpolateElement } from '@wordpress/element';
+import { useI18n } from '@wordpress/react-i18n';
+import { FormEvent, useEffect, useState } from 'react';
+import FormattedHeader from 'calypso/components/formatted-header';
+import { useUploadSiteIcon } from 'calypso/data/media/use-upload-site-icon';
+import { SITE_STORE } from 'calypso/landing/stepper/stores';
+import { recordTracksEvent } from 'calypso/lib/analytics/tracks';
+import { useSite } from '../../../../hooks/use-site';
+import SetupForm from '../link-in-bio-setup/setup-form';
+import type { Step } from '../../types';
+
+import '../link-in-bio-setup/styles.scss';
+
+const LinkInBioPostSetup: Step = function LinkInBioPostSetup( { navigation } ) {
+	const { submit } = navigation;
+	const { __ } = useI18n();
+	const site = useSite();
+	const uploadSiteIcon = useUploadSiteIcon();
+
+	const [ siteTitle, setComponentSiteTitle ] = useState( '' );
+	const [ tagline, setTagline ] = useState( '' );
+	const [ invalidSiteTitle, setInvalidSiteTitle ] = useState( false );
+	const [ selectedFile, setSelectedFile ] = useState< File | undefined >();
+	const [ base64Image, setBase64Image ] = useState< string | null >();
+
+	const { saveSiteSettings } = useDispatch( SITE_STORE );
+
+	useEffect( () => {
+		setComponentSiteTitle( site?.name || '' );
+		setTagline( site?.description || '' );
+	}, [ site ] );
+
+	const handleSubmit = async ( event: FormEvent ) => {
+		event.preventDefault();
+		setInvalidSiteTitle( ! siteTitle.trim().length );
+
+		if ( site ) {
+			await saveSiteSettings( site.ID, {
+				blogname: siteTitle,
+				blogdescription: tagline,
+			} );
+		}
+
+		if ( selectedFile && base64Image ) {
+			await uploadSiteIcon(
+				base64ImageToBlob( base64Image ),
+				'site-logo.png',
+				site?.ID,
+				site?.icon?.media_id,
+				site
+			);
```

Comment:
If you use this [hook](https://github.com/Automattic/wp-calypso/blob/trunk/packages/data-stores/src/queries/use-site-logo-mutation.ts#L23), the logo problem will go away. Because the hook changes `site_logo` **_and_** `site_icon`.

Reply:
Just a note that this has been done as well, and logo issue is resolved.